tests/phpunit/tests/ajax/CropImage.php

```diff
@@ -0,0 +1,158 @@
+		$attachment = $this->create_attachment( false );
+		$this->prepare_post( $attachment );
+
+		// Make the request.
+		try {
+			$this->_handleAjax( 'crop-image' );
+		} catch ( WPAjaxDieContinueException $e ) {
+		}
+
+		$response = json_decode( $this->_last_response, true );
+		$this->validate_response( $response );
+
+		$cropped_attachment = get_post( $response['data']['id'] );
+		$this->assertInstanceOf( WP_Post::class, $cropped_attachment, 'get_post function must return an instance of WP_Post class' );
+		$this->assertEmpty( $attachment->post_title, 'post_title value must be empty for testing purposes' );
+		$this->assertNotEmpty( $cropped_attachment->post_title, 'post_title value must be auto-generated if it\'s empty in the original attachment' );
+		$this->assertStringStartsWith( 'http', $cropped_attachment->post_content, 'post_content value should contain an URL if it\'s empty in the original attachment' );
+		$this->assertEmpty( $cropped_attachment->post_excerpt, 'post_excerpt value must be empty if it\'s empty in the original attachment' );
+		$this->assertEmpty( $cropped_attachment->_wp_attachment_image_alt, '_wp_attachment_image_alt value must be empty if it\'s empty in the original attachment' );
+
+		wp_delete_attachment( $attachment->ID, true );
+		wp_delete_attachment( $cropped_attachment->ID, true );
+	}
+
+	/**
+	 * Creates an attachment.
+	 *
+	 * @return WP_Post
+	 */
+	private function create_attachment( $with_metadata = true ) {
+		$uniq_id = uniqid( 'crop-image-ajax-action-test-' );
+		$object  = array(
+			'post_title'     => $with_metadata ? 'Title ' . $uniq_id : '',
+			'post_content'   => $with_metadata ? 'Description ' . $uniq_id : '',
+			'post_mime_type' => 'image/jpg',
+			'guid'           => 'http://localhost/foo.jpg',
+			'context'        => 'custom-logo',
+			'post_excerpt'   => $with_metadata ? 'Caption ' . $uniq_id : '',
+		);
+
+		$test_file        = DIR_TESTDATA . '/images/test-image.jpg';
+		$upload_directory = wp_upload_dir();
+		$uploaded_file    = $upload_directory['path'] . '/' . $uniq_id . '.jpg';
+		$filesystem       = new WP_Filesystem_Direct( true );
+		$filesystem->copy( $test_file, $uploaded_file );
+		$attachment_id = wp_insert_attachment( $object, $uploaded_file );
+
+		if ( $with_metadata ) {
+			update_post_meta( $attachment_id, '_wp_attachment_image_alt', wp_slash( 'Alt ' . $uniq_id ) );
+		}
+
+		return get_post( $attachment_id );
+	}
+
+	/**
+	 * @param array $response Response to validate.
+	 *
+	 * @return void
+	 */
+	private function validate_response( $response ) {
+		$this->assertArrayHasKey( 'success', $response, 'Response array must contain "success" key.' );
+		$this->assertArrayHasKey( 'data', $response, 'Response array must contain "data" key.' );
+		$this->assertNotEmpty( $response['data']['id'], 'Response array must contain "ID" value of the post entity.' );
+	}
+
+	/**
+	 * Prepares $_POST for crop-image ajax action.
+	 *
+	 * @param WP_Post $attachment
+	 *
+	 * @return void
```

Comment:
```suggestion
```

Core doesn't use `@return void`.

Reply:
Fixed in a4cb089d043fefcc7e957e13428f4e0da059ffbd